index.js

```diff
@@ -1,26 +1,46 @@
 const marked = require('marked');
 const TerminalRenderer = require('marked-terminal');
-const SemanticReleaseError = require('@semantic-release/error');
+const envCi = require('env-ci');
 const getConfig = require('./lib/get-config');
 const getNextVersion = require('./lib/get-next-version');
 const getCommits = require('./lib/get-commits');
 const logger = require('./lib/logger');
 const {gitHead: getGitHead, isGitRepo} = require('./lib/git');
 
 module.exports = async opts => {
+  const {isCi, branch, isPr} = envCi();
+
+  if (!isCi) {
+    logger.log('This run was not triggered in a CI environment, running in dry-run mode.');
+    opts.dryRun = true;
+  }
+
+  if (isCi && isPr) {
+    logger.log('This run was triggered by a pull request and therefore a new version won’t be published.');
+    return;
+  }
+
   if (!await isGitRepo()) {
-    throw new SemanticReleaseError('Semantic-release must run from a git repository', 'ENOGITREPO');
+    logger.error('Semantic-release must run from a git repository.');
+    return;
   }
 
   const config = await getConfig(opts, logger);
   const {plugins, options} = config;
 
+  if (branch !== options.branch) {
+    logger.log(
+      `This test run was triggered on the branch ${branch}, while semantic-release is configured to only publish from ${
+        options.branch
+      }, therefore a new version won’t be published.`
+    );
+    return;
+  }
+
   logger.log('Run automated release from branch %s', options.branch);
 
-  if (!options.dryRun) {
-    logger.log('Call plugin %s', 'verify-conditions');
-    await plugins.verifyConditions({options, logger});
-  }
+  logger.log('Call plugin %s', 'verify-conditions');
```

Comment:
why the `%s` placeholder here?

Reply:
It's to highlight `verify-conditions` with color. See [logger.js#L11](https://github.com/semantic-release/semantic-release/blob/caribou/lib/logger.js#L11).

In retrospective it's maybe unnecessary fancyness. But it's not related to this PR.